### PR TITLE
Refactoring - modern java

### PR DESCRIPTION
**Pull request by yanlyub**
refactored parts the code to Java 5 standards

### CODE REVIEW

File: src/VASSAL/tools/concurrent/listener/EventAccumulator.java [hunk 1]
```diff
@@ -19,10 +19,11 @@
 
 package VASSAL.tools.concurrent.listener;
 
+import org.apache.commons.lang3.tuple.ImmutablePair;
+
 import java.util.ArrayList;
 import java.util.Collection;
-
-import VASSAL.tools.lang.Pair;
+import java.util.List;
```
Comment (uckelman) [inline src/VASSAL/tools/concurrent/listener/EventAccumulator.java:26]: This import isn't used.

File: src/VASSAL/tools/imports/adc2/ADC2Module.java [hunk 2]
```diff
@@ -1736,9 +1736,9 @@ protected void readAllianceBlock(DataInputStream in) throws IOException {
     /* int nAlliances = */ ADC2Utils.readBase250Word(in); // ignored
     for (int i = 0; i < players.size(); ++i) {
       in.readUnsignedShort(); // unknown
-      for (int j = 0; j < players.size(); ++j) {
+      for (Player player : players) {
         if (in.readUnsignedShort() > 0) {
-          players.get(i).setAlly(players.get(j));
+          players.get(i).setAlly(player);
```
Comment (uckelman) [inline src/VASSAL/tools/imports/adc2/ADC2Module.java:1741]: Both loops could be replaced:
``` 
    for (Player p1 : players) {
      in.readUnsignedShort(); // unknown
      for (Player p2 : players) {
        if (in.readUnsignedShort() > 0) {
          p1.setAlly(p2);
        }
      }
    }
```

File: src/VASSAL/build/module/dice/BonesDiceServer.java [hunk 3]
```diff
@@ -54,14 +54,14 @@ public BonesDiceServer() {
...
     // format is "{{ xdy + n }}"
-    for (int i = 0; i < rolls.length; ++i) {
+    for (DieRoll roll : rolls) {
       query.append("{{")
-          .append(rolls[i].getNumDice())
-          .append("D")
-          .append(rolls[i].getNumSides());
+              .append(roll.getNumDice())
+              .append("D")
+              .append(roll.getNumSides());
```
Comment (uckelman) [inline src/VASSAL/build/module/dice/BonesDiceServer.java:61]: These should be realigned under the `.`.

File: src/VASSAL/build/module/gamepieceimage/InstanceConfigurer.java [hunk 4]
```diff
@@ -133,21 +133,17 @@ public static String PropertiesToString(List<ItemInstance> props) {
...
+    for (String item : p) {
+      if (item.startsWith(SymbolItem.TYPE)) {
+        props.add(new SymbolItemInstance(item, defn));
+      } else if (item.startsWith(TextBoxItem.TYPE)) {
+        props.add(new TextBoxItemInstance(item, defn));
+      } else if (item.startsWith(TextItem.TYPE)) {
+        props.add(new TextItemInstance(item, defn));
+      } else if (item.startsWith(ShapeItem.TYPE)) {
+        props.add(new ShapeItemInstance(item, defn));
+      } else if (item.startsWith(ImageItem.TYPE)) {
```
Comment (uckelman) [inline src/VASSAL/build/module/gamepieceimage/InstanceConfigurer.java:145]: `}` before `else if` should be on its own line.

File: src/VASSAL/counters/Deck.java [hunk 5]
```diff
@@ -1127,12 +1127,11 @@ public void actionPerformed(ActionEvent e) {
...
-    for (int i = 0; i < commands.length; ++i) {
-      if (Resources.getString("Deck.face_up").equals(commands[i].getValue(Action.NAME)) && !faceDown) { //$NON-NLS-1$
-        commands[i].putValue(Action.NAME, Resources.getString("Deck.face_down")); //$NON-NLS-1$
-      }
-      else if (Resources.getString("Deck.face_down").equals(commands[i].getValue(Action.NAME)) && faceDown) { //$NON-NLS-1$
-        commands[i].putValue(Action.NAME, Resources.getString("Deck.face_up")); //$NON-NLS-1$
+    for (KeyCommand command : commands) {
+      if (Resources.getString("Deck.face_up").equals(command.getValue(Action.NAME)) && !faceDown) { //$NON-NLS-1$
+        command.putValue(Action.NAME, Resources.getString("Deck.face_down")); //$NON-NLS-1$
+      } else if (Resources.getString("Deck.face_down").equals(command.getValue(Action.NAME)) && faceDown) { //$NON-NLS-1$
```
Comment (uckelman) [inline src/VASSAL/counters/Deck.java:1133]: `else if` must start line.

File: src/VASSAL/counters/RestrictCommands.java [hunk 6]
```diff
@@ -125,20 +125,19 @@ public Command keyEvent(KeyStroke stroke) {
...
           if (action.equals(DISABLE)) {
-            KeyCommand newCommand = new KeyCommand(commands[i]);
+            KeyCommand newCommand = new KeyCommand(command);
             newCommand.setEnabled(false);
             newCommands.add(newCommand);
           }
-        }
-        else {
-          newCommands.add(commands[i]);
+        } else {
```
Comment (uckelman) [inline src/VASSAL/counters/RestrictCommands.java:139]: `else` must start line.

File: src/VASSAL/i18n/LocaleConfigurer.java [hunk 7]
```diff
@@ -140,10 +140,10 @@ protected void updateValue() {
     if (languageList == null) {
       String[] langs = Locale.getISOLanguages();
       ArrayList<String> sortedLangs = new ArrayList<String>();
-      for (int i = 0; i < langs.length; i++) {
+      for (String s : langs) {
         String lang =
-          (new Locale(langs[i])).getDisplayLanguage(Locale.getDefault());
-        languages.put(lang, langs[i]);
+                (new Locale(s)).getDisplayLanguage(Locale.getDefault());
```
Comment (uckelman) [inline src/VASSAL/i18n/LocaleConfigurer.java:145]: Put this on the same line as the `String lang`, as it fits now.

File: src/VASSAL/launch/TilingHandler.java [hunk 8]
```diff
@@ -167,32 +158,30 @@ protected Dimension getImageSize(DataArchive archive, String ipath)
...
+            System.getProperty("java.class.path"),
+            "-Xmx" + maxheap + "M",
+            "-DVASSAL.id=" + pid,
+            "-Duser.home=" + System.getProperty("user.home"),
+            "-DVASSAL.port=" + port,
+            "VASSAL.tools.image.tilecache.ZipFileImageTiler",
+            aname,
+            cdir.getAbsolutePath(),
+            String.valueOf(tdim.width),
+            String.valueOf(tdim.height));
```
Comment (uckelman) [inline src/VASSAL/launch/TilingHandler.java:184]: Should be indented only two beyond line 172.

File: src/VASSAL/launch/TilingHandler.java [hunk 9]
```diff
@@ -209,20 +198,15 @@ protected void runSlicer(List<String> multi, final int tcount, int maxheap)
       null,
       outP,
       errP,
-      args.toArray(new String[args.size()])
+      args.toArray(new String[0])
```
Comment (uckelman) [inline src/VASSAL/launch/TilingHandler.java:201]: This looks wrong. Why create a new array which is the wrong size?

File: src/VASSAL/build/module/Map.java [hunk 10]
```diff
@@ -1580,17 +1580,16 @@ public void drawPiecesInRegion(Graphics g,
...
         getStackMetrics().draw(
-          (Stack) stack[i], pt, g, this, dzoom, visibleRect
+                (Stack) gamePiece, pt, g, this, dzoom, visibleRect
         );
-      }
-      else {
-        stack[i].draw(g, pt.x, pt.y, c, dzoom);
-        if (Boolean.TRUE.equals(stack[i].getProperty(Properties.SELECTED))) {
-          highlighter.draw(stack[i], g, pt.x, pt.y, c, dzoom);
+      } else {
```
Comment (uckelman) [inline src/VASSAL/build/module/Map.java:1589]: `else` should start line.

**Comment by uckelman** [comment]
a8ceca0658 is going to cause problems with VASL in particular, because it subclasses `TilingHandler` and uses `Pair`, and any other module which uses `Pair` in custom code. I'd prefer to have that commit in a separate PR. Some consideration is required for whether the benefit of it is worth the additional work it requires.

File: src/VASSAL/launch/TilingHandler.java (hunk 9 above)
Comment (yanlyub) [reply src/VASSAL/launch/TilingHandler.java:201]: It looks wrong and is counterintuitive, but this is the recommended way in modern Java as it helps the JVM do internal optimization, which it cannot do if it is given a presized array.

More details can be found here:
- https://stackoverflow.com/a/29444594
- https://code.i-harness.com/en/q/2a80d
- https://shipilev.net/blog/2016/arrays-wisdom-ancients/

**Comment by yanlyub** [comment]
- reverted a8ceca0
- applied the other requested changes
- adjusted my code formatter to use 2-space indent and else if on new line

File: src/VASSAL/build/module/dice/BonesDiceServer.java [hunk 11]
```diff
@@ -55,10 +55,11 @@ public BonesDiceServer() {
...
     for (DieRoll roll : rolls) {
-      query.append("{{")
-              .append(roll.getNumDice())
-              .append("D")
-              .append(roll.getNumSides());
+      query
+        .append("{{")
+        .append(roll.getNumDice())
+        .append("D")
+        .append(roll.getNumSides());
```
Comment (uckelman) [inline src/VASSAL/build/module/dice/BonesDiceServer.java:62]: No, not this. It should be:
```
query.append("{{")
     .append(roll.getNumDice())
     .append("D")
     .append(roll.getNumSides());
```

**Pull request merged**